src/main/java/org/folio/processing/mapping/mapper/reader/record/MarcRecordReader.java

```diff
@@ -1,61 +1,178 @@
+    List<String> resultList = new ArrayList<>();
+    for (String expression : expressions) {
+      StringBuilder sb = new StringBuilder();
+      String[] expressionParts = expression.split(WHITESPACE_DIVIDER);
+      for (String expressionPart : expressionParts) {
+        if (MARC_PATTERN.matcher(expressionPart).matches()) {
+          List<String> marcValues = readValuesFromMarcRecord(expressionPart);
+          if (arrayValue) {
+            resultList.addAll(marcValues);
+          } else {
+            marcValues.forEach(v -> sb.append(v));
+          }
+        } else if (STRING_VALUE_PATTERN.matcher(expressionPart).matches()) {
+          String value = expressionPart.replace("\"", "");
+          if (ruleExpression.getAcceptedValues() != null && !ruleExpression.getAcceptedValues().isEmpty()) {
+            for (Map.Entry<String, String> entry : ruleExpression.getAcceptedValues().entrySet()) {
+              if (entry.getValue().equals(value)) {
+                value = entry.getKey();
+              }
+            }
+          }
+          if (arrayValue) {
+            resultList.add(value);
+          } else {
+            sb.append(value);
+          }
+        }
+      }
+      resultList.remove(" ");
+      if (arrayValue && !resultList.isEmpty()) {
+        return ListValue.of(resultList);
+      }
+      if (isNotBlank(sb.toString())) {
+        return StringValue.of(sb.toString());
+      }
+    }
+    return MissingValue.getInstance();
   }
 
-  private String getCondition(String ruleExpression) {
-    // Stub implementation
-    return StringUtils.EMPTY;
+  private Value readRepeatableField(MappingRule ruleExpression) {
+    List<RepeatableSubfieldMapping> subfields = ruleExpression.getSubfields();
+    MappingRule.RepeatableFieldAction action = ruleExpression.getRepeatableFieldAction();
+    List<Map<String, Value>> repeatableObject = new ArrayList<>();
+    for (RepeatableSubfieldMapping subfieldMapping : subfields) {
+      Map<String, Value> object = subfieldMapping.getFields()
+        .stream()
+        .map(mappingRule -> new ImmutablePair<>(mappingRule.getPath(), mappingRule.getBooleanFieldAction() != null
+          ? BooleanValue.of(mappingRule.getBooleanFieldAction())
+          : readSingleField(mappingRule))
+        ).collect(Collectors.toMap(Pair::getLeft, Pair::getRight));
+      repeatableObject.add(object);
+    }
+    return RepeatableFieldValue.of(repeatableObject, ruleExpression.getRepeatableFieldAction(), ruleExpression.getPath());
   }
 
-  private Value readByCondition(String tag, String condition) {
-    // Stub implementation
-    Value result = MissingValue.getInstance();
-    JsonNode fieldNode = this.fieldsNode.findValue(tag);
-    if (fieldNode != null && fieldNode.isTextual()) {
-      result = StringValue.of(fieldNode.textValue());
+  private List<String> readValuesFromMarcRecord(String marcPath) {
+    List<VariableField> fields = marcRecord.getVariableFields(marcPath.substring(0, 3));
```

Comment:
I think that this number "3" can be extracted as a constant or please just left a comment over there. From my side, it is unclear - why it is 3? (or not 5, for example)